Server/RequestProcessing/add_friend_request.cpp

```diff
@@ -0,0 +1,81 @@
+#include "add_friend_request.h"
+
+AddFriendRequest::AddFriendRequest(QByteArray &request, DAL * d, QTcpSocket *s) 
+  : AbstractRequest(d, s)
+{
+  income_data_ = Parser::ParseAsFriendRequestInfo(request);
+  PrepareResponse();
+}
+
+void AddFriendRequest::PrepareResponse()
+{
+  //FriendRequestInfo ; 
+  // other_login
+  //id
+  //password
+  try {
+    Client sender_guy = database_->getClient(income_data_.id); // he actually loginned so uptodated data is in db
+    
+    if (QString::compare(sender_guy.GetUserPassword(), income_data_.password) == 0) {
```

Comment:
**Here example of how to simplify this method** 

    Client sender_guy = database_->getClient(income_data_.id); // he actually loginned so uptodated data is in db
	response_to_requester_ = (quint8)ServerRequests::FRIEND_REQUEST_FAILED;

    if (QString::compare(sender_guy.GetUserPassword(), income_data_.password) == 0 &&
			database_->Check_If_Client_exists_In_Db(income_data_.other_login)) {

        requested_guy = database_->getClient(income_data_.other_login);
        info_to_send.requester_ip = sender_guy.GetUserIp();
        info_to_send.requester_login = sender_guy.GetUserName();
        info_to_send.requester_port = sender_guy.GetUserPort();
        response_to_requester_ = (quint8)ServerRequests::FRIEND_REQUEST_SUCCEED;
    }
  }
  catch (UserNotFound& e) {
    response_to_requester_ = (quint8)ServerRequests::FRIEND_REQUEST_FAILED;
    qDebug() << e.what();
  }
}